hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheGetConfigOperation.java

```diff
@@ -50,18 +50,15 @@ public CacheGetConfigOperation(String name, String simpleName) {
     public void run()
             throws Exception {
         final ICacheService service = getService();
-        final CacheConfig cacheConfig = service.getCacheConfig(name);
+        CacheConfig cacheConfig = service.getCacheConfig(name);
         if (cacheConfig == null) {
             CacheSimpleConfig simpleConfig = service.findCacheConfig(simpleName);
             if (simpleConfig != null) {
                 try {
-                    CacheConfig cacheConfigFromSimpleConfig = new CacheConfig(simpleConfig);
-                    cacheConfigFromSimpleConfig.setName(simpleName);
-                    cacheConfigFromSimpleConfig.setManagerPrefix(name.substring(0, name.lastIndexOf(simpleName)));
-                    if (service.putCacheConfigIfAbsent(cacheConfigFromSimpleConfig) == null) {
-                        response = cacheConfigFromSimpleConfig;
-                        return;
-                    }
+                    cacheConfig = new CacheConfig(simpleConfig);
+                    cacheConfig.setName(simpleName);
+                    cacheConfig.setManagerPrefix(name.substring(0, name.lastIndexOf(simpleName)));
+                    service.putCacheConfigIfAbsent(cacheConfig);
```

Comment:
You assume the existing config will always be exactly the same as the one you just created from the simple cache config. Is this assumption always correct?